File: Python/day_two.py
def anagrams(s1, s2):
  s1_dict = {}
  for letter in s1:
    if letter not in s1_dict:
      s1_dict[letter] = 1
    else:
      s1_dict[letter] += 1
  s2_dict = {}
  for letter in s2:
    if letter not in s2_dict:
      s2_dict[letter] = 1
    else:
      s2_dict[letter] += 1

  return s2_dict == s1_dict

File: JS/dayTwo.js
const anagrams = (s1, s2) => {
    const s1Count = {};

    for (let letter of s1) {
      if (s1Count[letter] === undefined) {
        s1Count[letter] = 1;
      } else {
        s1Count[letter] += 1;
      }
    }
    for (let letter of s2) {
      if (s1Count[letter] === undefined) {
        return false;
      } else {
        s1Count[letter] -= 1;
      }
    }
    for (let item in s1Count) {
      if (!(s1Count[item] === 0)) {
        return false;
      }
    } return true;
  };
